internal/pkg/agent/cmd/upgrade.go

```diff
@@ -53,20 +58,73 @@ func newUpgradeCommandWithArgs(_ []string, streams *cli.IOStreams) *cobra.Comman
 	cmd.Flags().String(flagPGPBytes, "", "PGP to use for package verification")
 	cmd.Flags().String(flagPGPBytesURI, "", "Path to a web location containing PGP to use for package verification")
 	cmd.Flags().String(flagPGPBytesPath, "", "Path to a file containing PGP to use for package verification")
+	cmd.Flags().BoolP(flagForce, "", false, "Advanced option to force an upgrade on a fleet managed agent")
+	err := cmd.Flags().MarkHidden(flagForce)
+	if err != nil {
+		fmt.Fprintf(streams.Err, "error while setting upgrade force flag attributes: %s", err.Error())
+		os.Exit(1)
+	}
 
 	return cmd
 }
 
+type upgradeInput struct {
+	streams   *cli.IOStreams
+	cmd       *cobra.Command
+	args      []string
+	c         client.Client
+	agentInfo info.Agent
+}
+
 func upgradeCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error {
 	c := client.New()
-	return upgradeCmdWithClient(streams, cmd, args, c)
+	agentInfo, err := info.NewAgentInfoWithLog(cmd.Context(), "error", false)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve agent info while tring to upgrade the agent: %w", err)
+	}
+	input := &upgradeInput{
+		streams,
+		cmd,
+		args,
+		c,
+		agentInfo,
+	}
+	return upgradeCmdWithClient(input)
 }
 
-func upgradeCmdWithClient(streams *cli.IOStreams, cmd *cobra.Command, args []string, c client.Client) error {
-	version := args[0]
+func shouldUpgrade(cmd *cobra.Command, agentInfo info.Agent) (bool, error) {
+	if agentInfo.IsStandalone() {
+		return true, nil
+	}
+
+	if agentInfo.Unprivileged() {
+		return false, fmt.Errorf("upgrade command needs to be executed as root for fleet managed agents")
+	}
+
+	force, err := cmd.Flags().GetBool(flagForce)
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve command flag information while trying to upgrade the agent: %w", err)
+	}
+
+	if !force {
+		return false, unsupportedUpgrade
+	}
+
+	return true, nil
+}
```

Comment:
If I am not missing something this function can only return:
- true, nil
- false, <some error>

If that is the case the boolean return value is superfluous and you can just return error and check that for success in the caller; maybe the function could change name to something like `checkUpgradable()` and we can pass a boolean for the force flag instead of the whole cmd,  something like:
```suggestion
func checkUpgradable(force bool, agentInfo info.Agent) error {
	if agentInfo.IsStandalone() {
		return nil
	}

	if agentInfo.Unprivileged() {
		return fmt.Errorf("upgrade command needs to be executed as root for fleet managed agents")
	}

	if !force {
		return unsupportedUpgrade
	}

	return nil
}
```